src/Workspaces/Core/Portable/Diagnostics/DiagnosticAnalysisResult.cs

```diff
@@ -27,45 +27,27 @@ internal readonly struct DiagnosticAnalysisResult
     /// The set of documents that has any kind of diagnostics on it.
     /// </summary>
     public readonly ImmutableHashSet<DocumentId>? DocumentIds;
-    public readonly bool IsEmpty;
 
     /// <summary>
     /// Syntax diagnostics from this file.
     /// </summary>
-    private readonly ImmutableDictionary<DocumentId, ImmutableArray<DiagnosticData>>? _syntaxLocals;
+    private readonly ImmutableDictionary<DocumentId, ImmutableArray<DiagnosticData>> _syntaxLocals;
```

Comment:
now can't ever be null (since we never create a 'initial/default' instance), which cleans up a lot of stuff.